### DISCUSSION TITLE
Add `cursor-pointer` to the switch element

### DISCUSSION THREAD

**Discussion by timgavin** [livewire/flux#898] [Feature Requests]
I think adding the `cursor-pointer` class to the switch element would give a clearer indication that the switch will perform an action when clicked.

**Comment by MiguelLaforest** [comment]
Yes, I would also add segmented radio group to the request

**Comment by joshhanley** [comment]
@timgavin do you know if there are any accessibility guidelines for that?

**Comment by timgavin** [comment]
Yes: The W3C APG shows a pointer on [button](https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch-button/) and [checkbox](https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch-checkbox/) toggles:

>To help people with visual impairments identify the switch as an interactive element, the cursor is changed to a pointer when hovering over the switch.

**Reply by joshhanley** [reply]
@timgavin thanks! I would see this as an issue then instead of a feature request. As such I'm going to close this. Would you be able to open an issue and be sure to include all the information you've found in the issue, so I can get Caleb to have a look at it? Feel free to make it more generic like "add `cursor-pointer` for accessibility", so something like that, so it can cover multiple components. Thanks!

**Reply by timgavin** [reply]
Done! https://github.com/livewire/flux/issues/953

**Reply by joshhanley** [reply]
Thanks!

**Comment by timgavin** [comment]
Also, the pointer might be a good idea on segmented radio groups as the current cursor is a text cursor, which is really confusing. If not a pointer, at least `cursor-default` to clarify that you can't actually type in the element.

In fact, the segmented group on this page (Oldest|Newest|Top) uses a pointer :)

The [UK gov design guidelines](https://design-system.service.gov.uk/components/accordion/) also show a pointer over checkboxes, radios, tabs, and accordions, FWIW.

**Comment by joffreypersia** [comment]
I think it is good practice to add cursor-pointer to all elements that have interactivity (buttons, radios, selects, etc.).

I am wondering why it is not the case yet. What's the reason behind for the team ?

**Reply by jeffchown** [reply]
See @joshhanley 's comment https://github.com/livewire/flux/issues/1112#issuecomment-2669841120
and Tailwind doc: https://tailwindcss.com/docs/upgrade-guide#buttons-use-the-default-cursor

**Reply by joffreypersia** [reply]
Yes, but even the TailwindCSS fix (that I did when I migrated) doesn't overwrite the code of the select for example with the `cursor-default`. 

@joshhanley it shouldn't be cursor-default on buttons of interactive elements by default. 

![CleanShot 2025-03-15 at 15 49 06@2x](https://github.com/user-attachments/assets/c8abec24-2568-42e5-aca2-2425a796ebf3)

**Reply by jeffchown** [reply]
@joffreypersia No it doesn't/won't.  You would have to add CSS to target each specific use-case.

One of the reasons I linked Josh's comment in my earlier reply was to being attention to his last paragraph which explain's Flux's current position (for now) on this item/issue:

> We're not going to change this in Flux as we try to follow Tailwind's recommendations as much as possible, so I'm going to close this issue. If more people run into this, we will look at adding a note to the Flux upgrade guide.

I have mixed feelings about the Tailwind decision, in this case, and bet this change will be an ongoing discussion/debate here.

**Reply by joffreypersia** [reply]
Thank you for you answer. I think I am still "pro pointer" even though the theory says that is should be applied to links only because of their weak appearance of interactivity. 
But this is not the place for the debate x)

**Reply by joshhanley** [reply]
> @joshhanley it shouldn't be cursor-default on buttons of interactive elements by default.

@joffreypersia I don't exactly follow what you mean here. But if there is an issue stopping you overriding styles, please open an issue so we can investigate. Thanks!
